Basic example

orestesG opened the pull request:
Basic functionality issues trying angular 2

Commit:
angular package definition and configuration

Commit:
upgrading bootstrap version to the last one avilable

Commit:
simple navigation component mock up

Commit:
adding some service

Commit:
setting app for use HTTP Services and simulating WEB API

Commit:
trying angular HTTP library for simulate server request

Commit:
adding support for templates and applying some styles

jlulloav commented on app/category.services.ts:
```diff
@@ -0,0 +1,10 @@
+
+@Injectable()
+export class CategoryServices {
+    getCategoties() {
```
There is a typo here.

jlulloav commented on app/what-we-do.component.ts:
```diff
@@ -1,9 +1,24 @@
+export class WhatWeDo implements OnInit {
+    categories: Category[];
+    title = "What we do section";
+    constructor(private categoryService: CategoryServices) {
```
Typo

jlulloav commented on app/what-we-do.component.ts:
```diff
@@ -1,9 +1,24 @@
+
+    }
+    getCategories() {
+        this.categoryService.getCategoties().then(cat => this.categories = cat);
```
typo

jlulloav commented on app/main.ts:
```diff
@@ -1,7 +1,15 @@
 import { appRouterProviders } from "./app.routes";
 
-bootstrap(AppComponent,[appRouterProviders]);
+
```
These changes seems to be to communicate with an API. That's great!

Can you please let me know how it works? I would like to know the meaning of each line of code.

Thanks

orestesG replied:
This is the way  for register a router provider in the bootstrap method,  Angular provide   the Component Router service for present a particular component from a given URL, a router configuration is needed and that come from "app.routes" exports

orestesG replied:
Sorry, you have been talking about the http module, this module provide any type of http request , i think it might be useful simulate a web api, there is the api reference for this module https://angular.io/docs/ts/latest/api/http/index/Http-class.html

jlulloav commented on app/feedback.details.component.ts:
```diff
@@ -0,0 +1,24 @@
+    constructor(private feedbackServices: FeedbackServices) { }
+
+    getFeedback() {
+        this.feedbackServices.getFeedbacks().then(feedbck => this.feedbacks = feedbck)
```
there is a typo here

jlulloav commented on css/styles.css:
```diff
@@ -13,13 +13,14 @@ body {
 nav a {
```
Let's use SASS. http://sass-lang.com/documentation/file.SASS_REFERENCE.html

I will be the same at the end since the .scss gets converted to .css.

jlulloav commented:
Good job Orestes!

What we need to do.
1. See my comments. Let's fix those little stuffs.
2. I'm not seeing unit testing. How can we make sure everything is working fine? Let's add unit testing to the project.
3. Please update the readme file. Here is a guide https://dbader.org/blog/write-a-great-readme-for-your-github-project
4. Please take a look at the style guide. https://angular.io/docs/ts/latest/guide/style-guide.html

Thanks!

orestesG committed:
Update README.md

Commit:
some typo corrections

Commit:
enhancing syntax, conventions, and structuring

Commit:
fixing templates references path in respective components

Commit:
avoid using absolute path in urls for component html or css in the @Component
Note:
We decide on CommonJS formats AND we use a standard module loader, then we can use the module.id

Commit:
fixing the relative path for global style

Commit:
adding missing template

Commit:
including node-sass library for preprocessing Sass sheets

Commit:
adding some model an pipe test

Commit:
updating README.md file

Commit:
lower case contact page title for use capitalize-first pipe

Commit:
updating readme file about commands in scripts defined in the package.json

Commit:
deleting from repo compiled files, since are being compiled when  run the compiler at `npm run start`

Commit:
deleting from repo compiled files, since are being compiled when  run the compiler at `npm run start`

orestesG committed:
deleting from repo compiled files, since are being compiled when  run the compiler at `npm run start`

orestesG committed:
adding Karma  tool for real browser test

orestesG committed:
deleting from repo compiled files, since are being compiled when  run the compiler at `npm run start`

orestesG committed:
Merge branch 'master' into unit_test

orestesG committed:
updating readme file about Karma usage

Commit:
Adding End-to-end test configuration for Protractor framework

Commit:
Updating README.md file with ent-to-end spec

Commit:
Adding some basic e2e test

Commit:
Merge branch 'sass_compile'

Conflicts:
	package.json

orestesG committed:
Fixing readme.md subhead

Commit:
fixing build-css npm script output directory in package.json

Commit:
Merge branch 'master' of https://github.com/orestesG/lets-try-angular-2-typescript

Commit:
adding capability for simulate backend e.g submit Feedback
